server/src/main/java/org/candlepin/model/AbstractHibernateCurator.java

```diff
@@ -183,6 +183,20 @@ public E create(E entity) {
         return entity;
     }
 
+    /**
+     * @param entity to be created.
+     * @param flush whether or not to flush after the persist.
+     * @return newly created entity.
+     */
+    @Transactional
+    public E create(E entity, boolean flush) {
+        getEntityManager().persist(entity);
+        if (flush) {
+            flush();
+        }
+        return entity;
+    }
```

Comment:
Let's take this opportunity to clean up and sync our various methods here. Move the save method up next to these (or move these down to it), update the ```create(E)``` and ```save(E)``` (since create and save are synonyms) methods to call into this new create method with flush set to true. This way all of them will be doing the same thing, instead of having multiple implementations that all vary slightly.